### PR TITLE
Make the test project generator more composable

### PR DESCRIPTION
**Pull request by dac09**
Adds ability to
- use rufus (rwfw), I've called this flag `--link` but can rename to `--rwfw`
- clean target dir first
- overwrite existing target dir
- verbosely print every step
- setup tailwind with rufus

Also
- fixes blogpost codemod
- errors now cause failure in the script (fail fast and early)
- generally splits the code up into different files to make it easier to deal with.
- fixes https://github.com/redwoodjs/redwood/issues/3137

### CODE REVIEW

File: packages/cli/src/commands/dev.js [hunk 1]
```diff
@@ -93,8 +93,7 @@ export const handler = async ({
   const jobs = {
     api: {
       name: 'api',
-      command:
-        `yarn cross-env NODE_ENV=development NODE_OPTIONS=--enable-source-maps yarn nodemon --watch "${redwoodConfigPath}" --exec "yarn rw-api-server-watch"`,
+      command: `yarn cross-env NODE_ENV=development NODE_OPTIONS=--enable-source-maps yarn nodemon --watch "${redwoodConfigPath}" --exec "yarn rw-api-server-watch"`,
```
Comment (dac09) [inline packages/cli/src/commands/dev.js:96]: Just a lint fix

**Comment by dac09** [comment]
@thedavidprice done with changes. Plz murj if happy.

✅ Tailwind works
🐶 Rufus is a happy pup
🚨 Fails if any of the steps fail

**Comment by thedavidprice** [comment]
No time to review now. Merging and will go through later. Follow up PR if changes suggested.

Thanks!

File: packages/cli/src/commands/dev.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const builder = (yargs) => {
       description: 'Which dev server(s) to start',
       type: 'array',
     })
-    .positional('forward', {
+    .option('forward', {
```
Comment (thedavidprice) [inline packages/cli/src/commands/dev.js:22]: before I merge, can we think of any _good_ reason why this would have been a positional originally?

Comment (dac09) [reply packages/cli/src/commands/dev.js:22]: Just a typo/bug - we probably never noticed it because it runs headless in the workflow

**Pull request merged**